pkgs/misc/cups/libppd.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, cups
+, libcupsfilters
+, ghostscript
+, pkg-config
+, poppler_utils
+, mupdf # mutool
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libppd";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libppd";
+    rev = "2c650af8c8826a68c0c0293ab5511701eaac019a";
+    hash = "sha256-4KRJfzK/xeWlXmhUOlvGHsMmKqF6R4e5BiIA+h2UTII=";
+  };
+
+  configureFlags = [
+    "--disable-acroread"
+    "--localstatedir=/var"
+    "--sysconfdir=/etc"
+  ];
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    ghostscript
+    poppler_utils
+    mupdf.bin
+    libcupsfilters
+    zlib
+  ];
+
+  # Fix missing includes on darwin
+  prePatch = lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc")
+    ''
+      substituteInPlace ppd/ppd-collection.cxx \
+        --replace \
+        '#include <sys/wait.h>' \
+        "#include <sys/wait.h>''\n#include <signal.h>''\n#include <libgen.h>''\nextern char **environ;"
+
+      substituteInPlace Makefile.am \
+        --replace \
+        'libppd_la_LIBADD = ' \
+        'libppd_la_LIBADD = ''$(ZLIB_LIBS) ' \
+        --replace \
+        'libppd_la_CFLAGS = ' \
+        'libppd_la_CFLAGS = ''$(ZLIB_CFLAGS) '
+
+      substituteInPlace configure.ac \
+        --replace \
+        'PKG_CHECK_MODULES([LIBCUPSFILTERS], [libcupsfilters])' \
+        'PKG_CHECK_MODULES([LIBCUPSFILTERS], [libcupsfilters])''\nPKG_CHECK_MODULES([ZLIB], [zlib])'
+    '';
+
+  meta = with lib; {
+    description = "Legacy support library for PPD files";
+    homepage = "https://github.com/OpenPrinting/libppd";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ tmarkus ];
+    platforms = platforms.unix;
+  };
+}
+
```

Comment:
nit: you have two new lines after the final line of code, it should be one.

Reply:
resolved in rebased commits